silkworm/common/log.hpp

```diff
@@ -34,8 +35,8 @@ class Logger {
     int level() { return verbosity; }
 
     std::ostream& log(LogLevels level) {
-        std::time_t time = std::time(nullptr);
-        return stream << LogTags[level] << "[" << std::put_time(std::gmtime(&time), "%m-%d|%H:%M:%S") << "] ";
+        return stream << LogTags[level] << "["
+                      << absl::FormatTime("%m-%d|%H:%M:%E3S", absl::Now(), absl::LocalTimeZone()) << "] ";
```

Comment:
Per abseil documentation is preferrable to call FormatTime with an explicitly declared time zone.
Maybe setting `absl::TimeZone timezone;` and init on logger constructor ?

Reply:
OK.  @AlexeyAkhunov and I preferred UTC, Turbogeth notwithstanding. And my measures of C++ I/O showed it to be too slow for millisecond resolution to be meaningful.  Maybe I mismeasured.

Reply:
```
$ cat chrono.cpp
#include <iostream>
#include <chrono>

int main() 
{
    typedef std::chrono::system_clock clock_type;

    auto now = clock_type::now();
    auto seconds = std::chrono::time_point_cast<std::chrono::seconds>(now);
    auto fraction = now - seconds;
    time_t cnow = clock_type::to_time_t(now);

    auto milliseconds = std::chrono::duration_cast<std::chrono::milliseconds>(fraction);
    std::cout << "Milliseconds: " << milliseconds.count() << '\n';
    std::cout << "Milliseconds: " << milliseconds.count() << '\n';
    std::cout << "Milliseconds: " << milliseconds.count() << '\n';
    std::cout << "Milliseconds: " << milliseconds.count() << '\n';
    std::cout << "Milliseconds: " << milliseconds.count() << '\n';
}
Silver:build gcolvin$ g++ -std=c++11 -O3 chrono.cpp
Silver:build gcolvin$ ./a.out
Milliseconds: 56
Milliseconds: 56
Milliseconds: 56
Milliseconds: 56
Milliseconds: 56
```